### PR TITLE
fix: clipboard button

### PR DESCRIPTION
**Pull request by llorentelemmc**

### CODE REVIEW

File: AMW_angular/io/src/app/deployments/deployments.component.html [hunk 1]
```diff
@@ -12,6 +12,11 @@
       [message]="errorMessage"
       [messageType]="'warning'"
     ></app-notification>
+    <div *ngIf="showToast" class="toast position-absolute top-0 end-0" role="alert" aria-live="assertive" aria-atomic="true">
```
Comment (mburri) [inline AMW_angular/io/src/app/deployments/deployments.component.html:15]: This should probably go into a simple reusable component.

File: AMW_angular/io/src/app/deployments/deployments.component.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Location } from '@angular/common';
-import { Component, OnInit, ViewChild } from '@angular/core';
+import {ChangeDetectorRef, Component, OnInit, ViewChild} from '@angular/core';
```
Comment (mburri) [inline AMW_angular/io/src/app/deployments/deployments.component.ts:2]: ChangeDetectorRef seems to be unused and should be removed

File: AMW_angular/io/src/app/deployments/deployments.component.ts [hunk 3]
```diff
@@ -22,6 +22,13 @@ declare var $: any;
 @Component({
   selector: 'amw-deployments',
   templateUrl: './deployments.component.html',
+  styles: [`
```
Comment (mburri) [inline AMW_angular/io/src/app/deployments/deployments.component.ts:25]: If you create a dedicated toast component, this should go into this component too.

File: AMW_angular/io/src/app/deployments/deployments.component.ts [hunk 4]
```diff
@@ -289,22 +296,22 @@ export class DeploymentsComponent implements OnInit {
     );
   }
 
-  copyURL() {
+  showToast: boolean = false;
```
Comment (mburri) [inline AMW_angular/io/src/app/deployments/deployments.component.ts:299]: Like in java classes, property are usually declared at the top and not in between methods. Just move it up to the top of the class

File: AMW_angular/io/src/app/deployments/deployments.component.ts [hunk 5]
```diff
@@ -289,22 +296,22 @@ export class DeploymentsComponent implements OnInit {
...
-          '<input type="text" name="fname" class="textToCopyInput" style="opacity:0"/>'
-        ).val(url)
-      )
-      .find('.textToCopyInput')
-      .select();
     try {
-      document.execCommand('copy');
-    } catch (e) {
+      await navigator.clipboard.writeText(url);
+      console.log('URL copied to clipboard');
```
Comment (mburri) [inline AMW_angular/io/src/app/deployments/deployments.component.ts:305]: With the new toast and the window prompt if something fails, there is no need to log to the console.

File: AMW_angular/io/src/app/deployments/deployments.component.ts [hunk 6]
```diff
@@ -289,22 +296,22 @@ export class DeploymentsComponent implements OnInit {
...
+      this.showToast = true;
+      setTimeout(() => {
+        this.showToast = false;
+      }, 3000);
+
+    } catch (err) {
+      console.error('Failed to copy: ', err);
```
Comment (mburri) [inline AMW_angular/io/src/app/deployments/deployments.component.ts:312]: this console.log can be removed too.

**Comment by mburri** [comment]
Seems good to me. @yvespp What do you think of the new Toast component that notifies the user that the URL was successfully copied to the clipboard?

**Comment by yvespp** [comment]
Tested it and works. It's not very visible maybe it can be styled similar to the toast of the JSF app?
Example when a resource relation is removed:
image

**Comment by llorentelemmc** [comment]
> Tested it and works. It's not very visible maybe it can be styled similar to the toast of the JSF app? Example when a resource relation is removed: image

Screenshot from 2023-11-06 17-06-08
@yvespp I've tried to style it more similarly to the notification of the JSF App, is it better like this?

**Comment by yvespp** [comment]
@llorentelemmc would it be possible to add the blue part with the exclamation mark? If it's to complicated it's ok like this.

**Review by yvespp — APPROVED** [review]

**Comment by llorentelemmc** [comment]
@yvespp I've added the blue part with the exclamation part and adjusted the colours a bit. What do you think?
Screenshot from 2023-11-09 17-20-23

**Comment by yvespp** [comment]
Cool thanks!

**Pull request merged**